optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -290,13 +293,37 @@ def __call__(self, trial):
             )
         )
 
+        trial.set_user_attr("elapsed_secs", elapsed_secs)
+        trial.set_user_attr("average_iteration_time", average_iteration_time)
+        trial.set_user_attr("step_name", self.step_name)
+
         self.trial_count += 1
 
         return val_score
 
 
 class LightGBMTuner(BaseTuner):
-    """Hyperparameter-tuning with Optuna for LightGBM."""
+    """Hyperparameter-tuning with Optuna for LightGBM.
+
+    Arguments and keyword arguments for `lightgbm.train()
+    <https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.train.html>`_ can be passed.
+    The arguments that only :class:`~optuna.integration.lightgbm.LightGBMTuner` has are listed
+    below:
+
+    Args:
+        time_budget:
+            A time budget for parameter tuning in seconds.
+        best_params:
+            A dictionary to store the best parameters.
+        tuning_history:
+            A List to store the history of parameter tuning.
+        study:
+            A :class:`~optuna.study.Study` instance to store optimization results. The
+            :class:`~optuna.trial.Trial` instances in it has the following user attributes:
+            ``elapsed_secs`` is the elapsed time since the optimization starts.
+            ``average_iteration_time`` is the average time of iteration to train the booster
+            model in the trial.
+    """
```

Comment:
How about documenting `__init__` in `__init__`, following this recent docstring rule https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#docstrings ?